admin/lib/validator/check-window-v2/check-window-add-validator.js

```diff
@@ -0,0 +1,204 @@
+'use strict'
+const R = require('ramda')
+
+const checkWindowErrorMessages = require('../../errors/check-window-v2')
+const dateService = require('../../../services/date.service')
+const monitor = require('../../../helpers/monitor')
+const ValidationError = require('../../validation-error')
+
+const checkWindowNameValidator = require('./check-window-name-validator')
+const dateValidator = require('../common/date-validator')
+const DateValidationData = require('../common/DateValidationData')
+
+const checkWindowAddValidator = {}
+
+/**
+ * Validates check window insertion data
+ * @param {Object} checkWindowData
+ * @returns {Object}
+ */
+checkWindowAddValidator.validate = (checkWindowData) => {
+  const validationError = new ValidationError()
+  const checkWindowName = R.path(['checkWindowName'], checkWindowData)
+  checkWindowNameValidator.validate(validationError, checkWindowName)
+
+  const {
+    adminStartDay,
+    adminStartMonth,
+    adminStartYear,
+    adminEndDay,
+    adminEndMonth,
+    adminEndYear,
+    familiarisationCheckStartDay,
+    familiarisationCheckStartMonth,
+    familiarisationCheckStartYear,
+    familiarisationCheckEndDay,
+    familiarisationCheckEndMonth,
+    familiarisationCheckEndYear,
+    liveCheckStartDay,
+    liveCheckStartMonth,
+    liveCheckStartYear,
+    liveCheckEndDay,
+    liveCheckEndMonth,
+    liveCheckEndYear
+  } = checkWindowData
+
+  const adminStartDateData = new DateValidationData()
+  adminStartDateData
+    .day(adminStartDay)
```

Comment:
I think this is 💯 much better.  Many thanks for the change!

Reply:
@jon-shipley thanks for the extensive review